src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Guid.cs

```diff
@@ -0,0 +1,100 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Text;
+
+namespace System.Text.Json
+{
+    public ref partial struct Utf8JsonWriter
+    {
+        /// <summary>
+        /// Writes the <see cref="Guid"/> value (as a JSON string) as an element of a JSON array.
+        /// </summary>
+        /// <param name="value">The value to be written as a JSON string as an element of a JSON array.</param>
+        /// <exception cref="JsonWriterException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteStringValue(Guid value)
+        {
+            if (_writerOptions.Indented)
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingValue();
+                }
+                WriteStringValueIndented(value);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingValue();
+                }
+                WriteStringValueMinimized(value);
+            }
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.String;
+        }
+
+        private void WriteStringValueMinimized(Guid value)
+        {
+            // Calculated based on the following: ',"Guid value"'
+            int bytesNeeded = JsonConstants.MaximumGuidLength + 1;
+            if (_buffer.Length < bytesNeeded)
+            {
+                GrowAndEnsure(bytesNeeded);
+            }
+
+            int idx = 0;
+            if (_currentDepth < 0)
+            {
+                _buffer[idx++] = JsonConstants.ListSeperator;
+            }
+
+            _buffer[idx++] = JsonConstants.Quote;
+
+            Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out int bytesWritten);
+            idx += bytesWritten;
+
+            _buffer[idx++] = JsonConstants.Quote;
+
+            Advance(idx);
+        }
+
+        private void WriteStringValueIndented(Guid value)
+        {
+            int idx = 0;
+            if (_currentDepth < 0)
+            {
+                while (_buffer.Length <= idx)
+                {
+                    GrowAndEnsure();
+                }
+                _buffer[idx++] = JsonConstants.ListSeperator;
+            }
+
+            if (_tokenType != JsonTokenType.None)
+                WriteNewLine(ref idx);
+
+            int indent = Indentation;
+            while (true)
+            {
+                bool result = JsonWriterHelper.TryWriteIndentation(_buffer.Slice(idx), indent, out int bytesWritten);
+                idx += bytesWritten;
+                if (result)
+                {
+                    break;
+                }
+                indent -= bytesWritten;
+                AdvanceAndGrow(idx);
+                idx = 0;
+            }
```

Comment:
This again looks like the same code that's in a bunch of WriteNumberValueIndented methods.  It should be refactored to avoid the ton of duplication.